pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -1510,16 +1518,16 @@ func (c *Cloud) CreateDisk(volumeOptions *VolumeOptions) (string, error) {
 
 	case VolumeTypeIO1:
 		// See http://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_CreateVolume.html for IOPS constraints
-		if volumeOptions.IOPSPerGB <= 0 || volumeOptions.IOPSPerGB > 30 {
-			return "", fmt.Errorf("invalid iopsPerGB value %d, must be 0 < IOPSPerGB <= 30", volumeOptions.IOPSPerGB)
+		if volumeOptions.IOPSPerGB <= 0 || volumeOptions.IOPSPerGB > MaxIOPSPerGB {
```

Comment:
I think we need to document other things too. For example, io1 requires a
minimum volume size 4GB. So when I created a class for io1 and requested a
3GB volume for that class in my claim, the claim stayed in a pending state.
When I ran "describe" on the claim, it told me that I needed to request at
least a 4GB volume. When the system gets in that state, there is no way to
ever fulfill the claim. So I had to delete the claim and submit a new one
with 4GB.

On Wed, Aug 24, 2016 at 9:07 AM, Jan Šafránek <notifications@github.com>
wrote:

> In pkg/cloudprovider/providers/aws/aws.go
> <https://github.com/kubernetes/kubernetes/pull/31115#discussion_r76052210>
> :
>
> > @@ -1510,16 +1518,16 @@ func (c *Cloud) CreateDisk(volumeOptions *VolumeOptions) (string, error) {
> >
> >  	case VolumeTypeIO1:
> >  		// See http://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_CreateVolume.html for IOPS constraints
> > -		if volumeOptions.IOPSPerGB <= 0 || volumeOptions.IOPSPerGB > 30 {
> > -			return "", fmt.Errorf("invalid iopsPerGB value %d, must be 0 < IOPSPerGB <= 30", volumeOptions.IOPSPerGB)
> > +		if volumeOptions.IOPSPerGB <= 0 || volumeOptions.IOPSPerGB > MaxIOPSPerGB {
>
> Similarly, requesting 100 IOPS/GB, AWS throws "InvalidParameterValue: Iops
> to volume size ratio of 100.000000 is too high; maximum is 50", so I will
> remove this check from this PR.
>
> —
> You are receiving this because you are on a team that was mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/31115/files/fd1075ce30c1f5cfd8a230500d92a1a4cb4de5ad#r76052210>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADDZzLmSfV7hZt4ck42eWsKaKm7pfrp5ks5qjEIGgaJpZM4Jp4sL>
> .
>